### PR TITLE
fix: Set IGNORE value in mask to -torch.inf

### PR DESCRIPTION
**Pull request by connor-henderson**
# Description

Supersedes https://github.com/neelnanda-io/TransformerLens/pull/319. Just copied the changes + the final requested change.

From the other PR description:
"Fixes https://github.com/neelnanda-io/TransformerLens/issues/318 - In pythia-70m, attn_scores can get below -1e5 (for some reason...). Masked attn scores are set to -1e5, so this makes the model attend outside the mask! This is dumb, and so I set things to -torch.inf instead. It's still registered as a buffer so should move to device + dtypes accordingly"

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### CODE REVIEW

**Comment by jbloomAus** [comment]
@connor-henderson looks good but can you please write a few quick tests for this? (I think it's possible to load pythia-70m in the tests)

**Comment by UFO-101** [comment]
Bump @jbloomAus . This looks ready to merge

**Comment by jbloomAus** [comment]
Thanks, you're right! Appreciate the contribution.

**Comment by jbloomAus** [comment]
@connor-henderson this is causing the test in main to fail, can you please take a look? I think the issue is that something you did failed once it was merged (hence the tests didn't pick it up). I think Bryce was looking at this but ultimately didn't find a solution.

**Comment by UFO-101** [comment]
@jbloomAus I've fixed this in my PR here #386 
> Fix bug caused by previous PR https://github.com/neelnanda-io/TransformerLens/pull/366. Applying softmax to a vector of -torch.infs returns nans. So we zero any nan values.
```
pattern = torch.where(torch.isnan(pattern), torch.zeros_like(pattern), pattern)
```

**Comment by jbloomAus** [comment]
Cheers, I'll make a PR to put this in so it isn't tied up with yours @UFO-101

**Comment by connor-henderson** [comment]
>  I've fixed this in my PR here #386
> 
> > Fix bug caused by previous PR #366. Applying softmax to a vector of -torch.infs returns nans. So we zero any nan values.
> 
> ```
> pattern = torch.where(torch.isnan(pattern), torch.zeros_like(pattern), pattern)
> ```
@jbloomAus @UFO-101 
Alternatively we could use `torch.iinfo(torch.int64).min # ~ -9.22337e+18` instead of `-torch.inf`, but replacing the NaNs with 0s works too
